azure/defaults.go

```diff
@@ -56,6 +56,12 @@ const (
 	DefaultImagePublisherID = "cncf-upstream"
 	// LatestVersion is the image version latest.
 	LatestVersion = "latest"
+	// DefaultPublicGalleryName is the default Azure Compute Gallery.
+	DefaultPublicGalleryName = "ClusterAPI-f72ceb4f-5159-4c26-a0fe-2ea738f0d019"
```

Comment:
Parroting @jsturtevant, are you planning to open a k8s.io PR to add this new gallery to the Terraform?

Reply:
Yes indeed, but I haven't started that yet. That's the third piece of the puzzle, in addition to this and kubernetes-sigs/image-builder#1578.

Reply:
does this gallery name relate to anything in here?

https://github.com/kubernetes/k8s.io/pull/7461

Reply:
It doesn't--I don't think there's a way to recreate a community gallery with the same unique name if it were to be deleted. It gets created for you, starting with your specified prefix, when you actually share the gallery to the world.

So AFAICT it isn't / shouldn't be part of the terraform for the gallery.

Reply:
This implies if the gallery were to be deleted, we'd have to change code to point to a new gallery with the same name prefix. That's not great. I can't add a "Delete" lock on the resource, because that has the side effect of disallowing new image definitions to be added or old versions to be deleted, etc.

Reply:
It's already got a unique name by virtue of me recreating everything in the CNCF community sub and then reverse-engineering the terraform. AFAICT, the unique name it gets is stored under its `sharingProfile` which isn't in the terraform I generated...but maybe can be added?

```shell
% az sig show -g cluster-api-gallery --gallery-name community_gallery -o yaml
description: Shared image gallery for Cluster API Provider Azure
id: /subscriptions/46678f10-4bbb-447e-98e8-d2829589f2d8/resourceGroups/cluster-api-gallery/providers/Microsoft.Compute/galleries/community_gallery
identifier:
  uniqueName: 46678f10-4bbb-447e-98e8-d2829589f2d8-COMMUNITY_GALLERY
location: northcentralus
name: community_gallery
provisioningState: Succeeded
resourceGroup: cluster-api-gallery
sharingProfile:
  communityGalleryInfo:
    communityGalleryEnabled: true
    eula: https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/LICENSE
    publicNamePrefix: ClusterAPI
    publicNames:
    - ClusterAPI-f72ceb4f-5159-4c26-a0fe-2ea738f0d019
    publisherContact: az-k8s-up-infra@microsoft.com
    publisherUri: https://github.com/kubernetes-sigs/cluster-api-provider-azure
  groups: null
  permissions: Community
sharingStatus: null
softDeletePolicy: null
tags:
  DO-NOT-DELETE: UpstreamInfra
  DateCreated: 10/24/2024
  creationTimestamp: '2024-10-24T17:36:37Z'
  jobName: image-builder-sig-ubuntu-2404
type: Microsoft.Compute/galleries
```

Reply:
But from the point of view of CAPZ's code here, this is the unique identifier we need to access images and it shouldn't change except in a disaster recovery case where we had to stand up a new community gallery.

Reply:
I could try to refactor this away from being a constant into something created at runtime that the user could override if needed. (Or I can follow up with that change after this merges.)

Reply:
If users can set the gallery name in the API, I don't think we need a separate toggle to change the default for now.